src/function/util.rs

```diff
@@ -12,6 +12,17 @@ use std::{
 
 use crate::{function::register_remove_handlers, trim_os_str};
 
+/// Write an optional value to the function directory if not None.
+macro_rules! write_opt {
+    ($dir:expr, $name:expr, $value:expr) => {
+        if let Some(value) = $value {
+            $dir.write($name, value.to_string())?;
+        }
+    };
+}
+
+pub(super) use write_opt;
+
```

Comment:
Not sure if you'd prefer a method here rather than is macro. I added it whilst also working on uac2 since that has a lot of `Option`: https://github.com/tuna-f1sh/usb-gadget/commit/b1221aa3e28eb5e9fc9b05801ab0f4714f10bfbe#diff-0ffa4f627a73792703fd6aad03203cb38670ba4088415c71694fb50ea5c72bb7R119

Reply:
I prefer doing it explicitly in this crate because sometimes you need decimals and sometimes you need hex.